nemo/collections/tts/torch/data_ssl_vocoder.py

```diff
@@ -0,0 +1,590 @@
+import json
```

Comment:
Seemed to be getting big so created a separate file. I can move it to data.py if you suggest doing so

Reply:
Yeah, let's keep all the TTS Datasets together for now--if it gets too bulky we can separate them.

It might be nice especially since there is some shared functionality with the `TTSDataset`, so grouping the two physically closer together makes sense to me.

Reply:
Hi Jocelyn! I now remember why this is a separate file. There were some cyclic dependency issues - ssl_tts.py import data.py and for this dataset in data_ssl_vocoder.py, I needed to import ssl_tts.py . Not sure what's the best way to resolve this dependency. I can work on a dataset class that does not require ssl_tts.py and only uses cached embeddings (instead of computing them at run time). This might take a little longer though. Let me know what you think might be the best.

Reply:
Ahh, I see. Would this be resolved by doing something like this in `ssl_tts.py` instead of importing the Dataset directly? https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/models/fastpitch.py#L533

Reply:
So now I have removed data_ssl_vocoder.py and created a new class in data.py FastPitchSSLDataset. This dataset requires supplementary data created with make_supdata.py and does not compute embeddings at run time (doe not require SSL model checkpoint). Made this change because for all training runs during the internship I first ran make_supdata.py (since it runs on GPU and is much faster than computing embeddings, pitch contours and mel specs at run time). This also reduces the size of the dataset class and does not require too many config parameters (Most config params are only needed while running make_supdata.py). I will look at make_supdata.py soon and clean it up and add some comments. Please let me know what you think.

Reply:
Took a quick look, looks good! Should be fine to have users do the preprocessing first.